xobjects/specialize_source.py

```diff
@@ -111,6 +111,15 @@ def specialize_source(source, specialize_for, search_in_folders=[]):
             "cuda": " ",
         }[specialize_for],
     )
+    newfilecontent = newfilecontent.replace(
+        "/*gpulocmem*/",
+        {
+            "cpu_serial": " ",
+            "cpu_openmp": " ",
+            "opencl": " ",
+            "cuda": " __device__ ",
```

Comment:
This I introduced not because of the shared memory but because of a device variable in the Bhabha header here:
https://github.com/pkicsiny/xfields/blob/main/xfields/headers/bhabha_spectrum.h#L167

I could not use the /*gpufun*/ flag because it adds a static inline on the CPU and that only works for functions and for variables this crashes.

The shared memory is only used in here:
https://github.com/pkicsiny/xfields/blob/main/xfields/headers/compute_slice_moments.h#L177
in one of the xfields headers.

Reply:
Make sense!

The guard here https://github.com/pkicsiny/xfields/blob/main/xfields/headers/compute_slice_moments.h#L7 in theory allows this being used on OpenCL, but I see that you make sure to restrict the kernel to only CUDA in python so that's good.

Reply:
That's true. Just for safety I have made a new commit to xfields where I now removed the opencl flag and renamed the guard to XFIELDS_COMPUTESLICEMOMENTS_CUDA.

Reply:
removed this